pdesolvers/heat_equation_1d/solver/explicit_solver.py

```diff
@@ -1,32 +1,59 @@
 import numpy as np
+from matplotlib import pyplot as plt
 
-from pdesolvers.heat_equation_1d.strategy.solver_strategy import SolverStrategy
+from pdesolvers.heat_equation_1d.heat_equation import HeatEquation
+from pdesolvers.heat_equation_1d.solver.solver_strategy import SolverStrategy
 
 
 class HeatEquationExplicitSolver(SolverStrategy):
 
-    def solve(self, context):
 
-        x = context._generate_x_grid()
+    def solve(self, equation : HeatEquation):
+
+        self.equation = equation
```

Comment:
don't think this is the best way to go about with this - i did this so that i could access the equation in the plot() function

i was initially gonna do an init so that both solve and plot has access to the equation, but then it would be like:

solver = Solver(HeatEquationCNSolver(equation))
solver.solve().plot()

and not 

solver = Solver(HeatEquationCNSolver())
solver.solve(equation).plot()

maybe im overcomplicating things